### ISSUE TITLE
debian bullseye: python packages missing from buildserver

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#4082]
@dev2718 said that it could be a wrong hard-coded swig3.0 reference. This was needed for older Debian versions but with new Debian versions it leads to not building the python binding (as swig3.0 is not available).

```
doc/COMPILE.md:-DSWIG_EXECUTABLE=/usr/bin/swig3.0
doc/COMPILE.md:For Bindings swig3 is recommended. swig2 only works on some distributions.
doc/COMPILE.md:Thus you need to install swig3 manually:
scripts/packaging/package.sh:   -DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
scripts/docker/debian/buster/Dockerfile:        swig3.0 \
scripts/docker/debian/stretch/Dockerfile:        swig3.0 \
scripts/docker/ubuntu/focal/Dockerfile:RUN ln -s /usr/bin/swig /usr/bin/swig3.0
scripts/docker/legacy/Dockerfile:       swig3.0 \
scripts/dev/configure-home:cmake -DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
scripts/dev/configure-debian-log:cmake -DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
scripts/dev/configure-debian-debug:cmake -DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
scripts/dev/configure-debian-optimizations:cmake -DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
scripts/dev/configure-ini:cmake -DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
src/bindings/swig/python/README.md:-DSWIG_EXECUTABLE=/usr/bin/swig3.0 \
```

**Comment by markus2330** [comment]
> How does "redshift" react when there is no "elektra" present on a system?

As the PR is now it will fail in `./configure`.

> For example "redshift" is in the official Debian repository, but "elektra" not. How is "redshift" packed then?

Elektra will have to be added to Debian's official repository (again).

> mention the "Install" (https://www.libelektra.org/installation/installation) section in the "Get started" (https://www.libelektra.org/getstarted/guide) section.

There is a link to doc/INSTALL.md in doc/GETSTARTED.md, can you maybe create an issue or PR with the text you would have expected?

**Comment by 4ydan** [comment]
### FLOSS H1

I m not sure If I did the right triage as I did not fully understand the problem. So I tried building elektra on debian and buster and checked if the python binding was included.

## Debian buster 
### 1. start docker container

### 2.
```
sudo apt get update
```
### 3.
```
sudo apt-get install cmake git build-essential
```
### 4.
```
git clone https://github.com/ElektraInitiative/libelektra.git
```
### 5.
```
cd libelektra && mkdir build
cd build
cmake ..
```

Log after cmake ..
```
-- Exclude binding python because neither swig2/3/4 found. Please install swig 2/3/4 and set -DSWIG_EXECUTABLE=
-- Exclude binding lua because neither swig2/3/4 found. Please install swig 2/3/4 and set -DSWIG_EXECUTABLE=
```

### 6.
```
sudo apt-get install swig
```

Log after cmake ..
```
-- Exclude binding python because python3 libs not found
```

### 7.
```
sudo apt-get install libpython3-dev
```

Log after cmake ..

```
-- Include binding python
CMake Warning at src/bindings/swig/python/CMakeLists.txt:94 (message):
  Python bindings for elektra tools require SWIG 4 or later.  Not building.


Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'distutils.sysconfig'
CMake Error at src/bindings/swig/python/CMakeLists.txt:126 (install):
  install DIRECTORY given no DESTINATION!
```


## Debian bullseye

### 1. start docker container

### 2.
```
sudo apt get update
```
### 3.
```
sudo apt-get install cmake git build-essential
```
### 4.
```
git clone https://github.com/ElektraInitiative/libelektra.git
```
### 5.
```
cd libelektra && mkdir build
cd build
cmake ..
```

Log after cmake ..:
```
-- Exclude binding python because neither swig2/3/4 found. Please install swig 2/3/4 and set -DSWIG_EXECUTABLE=
```


## Expected Result
debian buster and bullseye to have swig3
the python binding should be included

## Actual Result
swig 3 is not included in debian buster or bullseye so the python binding failed on building.

## System Information
docker container from windows
mcr.microsoft.com/vscode/devcontainers/base:0-buster
mcr.microsoft.com/vscode/devcontainers/base:0-bullseye

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
Swig is present in "all" Debian version.
https://tracker.debian.org/pkg/swig

Not sure what the problem is here.

**Comment by 4ydan** [comment]
So I build elektra on a debian bullseye OS by following https://www.libelektra.org/installation/installation. How can I help to solve this Issue. What exactly is the problem with swig? Should I just remove the hardcoded appearances from the files above mentioned?
Thx in advance

**Comment by markus2330** [comment]
> Should I just remove the hardcoded appearances from the files above mentioned?

It is the best guess so far what the problem could be. Even if it does not fully solve the issue, these references are not correct anymore.

**Comment by mpranj** [comment]
Closed via #4261.

**Comment by markus2330** [comment]
Thank you, works perfectly now! :fireworks: 

```
python3
>>> import kdb
```

From my point of view the release can be done asap.
